survey-sdk/src/main/java/co/nimblehq/survey/sdk/SurveyApi.kt

```diff
@@ -0,0 +1,88 @@
+package co.nimblehq.survey.sdk
+
+import co.nimblehq.survey.sdk.api.AppService
+import co.nimblehq.survey.sdk.entity.SurveyEntity
+import co.nimblehq.survey.sdk.network.NetworkBuilder
+import kotlinx.coroutines.*
+
+class SurveyApi private constructor() : NetworkBuilder() {
+    private val service: AppService by lazy { buildService() }
+    private var version = "v1"
+
+    companion object {
+        private val instance: SurveyApi by lazy { SurveyApi() }
+    }
+
+    fun withMode(debugMode: Boolean): SurveyApi {
+        setDebugMode(debugMode)
+        return this
+    }
+
+    fun withBaseUrl(url: String): SurveyApi {
+        setBaseUrl(url)
+        return this
+    }
+
+    fun withConnectionTimeout(timeout: Long): SurveyApi {
+        setConnectionTimeoutInSecond(timeout)
+        return this
+    }
+
+    fun withReadTimeout(timeout: Long): SurveyApi {
+        setReadTimeoutInSecond(timeout)
+        return this
+    }
+
+    fun withVersion(version: String): SurveyApi {
+        this.version = version
+        return this
+    }
+
+    //Below are public apis
```

Comment:
this is still an invalid comment because of no affected class, method or line of code 🤔 I think we don't need this comment